src/libs/lodepng/lodepng.c

```diff
@@ -5349,14 +5348,23 @@ unsigned lodepng_decode(unsigned char ** out, unsigned * w, unsigned * h,
             return 56; /*unsupported color mode conversion*/
         }
 
-        outsize = lodepng_get_raw_size(*w, *h, &state->info_raw);
-        *out = (unsigned char *)lodepng_malloc(outsize);
-        if(!(*out)) {
+        lv_draw_buf_t * new_buf = lv_draw_buf_create(*w, *h, LV_COLOR_FORMAT_ARGB8888, 4 * *w);
+        if(new_buf == NULL) {
             state->error = 83; /*alloc fail*/
+            return state->error;
```

Comment:
should we free out buffer as it maybe not NULL in decodeGeneric and set it to NULL ?

Reply:
Fixed memory leak in case decoding fails.